src/lib/operators/join_sort_merge.cpp

```diff
@@ -139,6 +139,24 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractReadOnlyOperatorImpl {
   std::vector<std::shared_ptr<RowIDPosList>> _output_pos_lists_left;
   std::vector<std::shared_ptr<RowIDPosList>> _output_pos_lists_right;
 
+  struct RowHasher {
+    size_t operator()(const RowID& row) const {
```

Comment:
For eight byte I wouldn't use a reference. But I don't have numbers to support that.